src/array-schema.ts

```diff
@@ -84,7 +84,7 @@ export class ArraySchema extends TypeSchema<"array"> {
         enum: [i]
       };
     });
-    if (schema.items && schema.items instanceof Schema) this.items = schema.items.compile().additionalProperties as PropertySchema;
+    if (schema.items && schema.items instanceof Schema) this.items = schema.items.compile();
```

Comment:
Just to see if I understand this at all 😄 

It seems this line is giving us the ability to delegate to the compile method which properties to return. Which lets the compile method for the DictionarySchema select not to use the additionalProperties property...

Reply:
Yes, that's correct. Should have never been part of array-schema logic but I didn't see it at the time.